correlation-one/openedx-edx-platform-poc
# Add Authentication middleware to our OEX PoC

## Description

This PR introduces the changes required to run authentication using a Auth0 JWT bearer token as provided by an external application such as the Fellow applications (training-client + training-server).

The `middleware` introduced in the django framework does the following:

- verifies if the bearer authentication header is present
- validates the bearer token
- fetches from Auth0 the user basic information
- checks if the same user email is a registered user
- if the user is not present, then it creates one

## Supporting information

Relates to: LET-22
Notion document: https://www.notion.so/correlation-one/OpenEdx-Authentication-1395babc1294492d8a16ea0de81e30bf

## Testing instructions

1. Start both training-client and training-server on your local development machine.
2. Log in as a valid user. We tipically use: `jerryseinfeld@event1.com` as a test user. This user is a registered user in Auth0.
3. Use your web browser tools to get from the local storage, the `access` token. Copy the token.
4. Initialize an Openedx `devstack` and make sure it is working.
5. Replace `edx-platform` with this fork and position your head branch to this one. Re-provision and make sure once more, the `lms` is up and running.
6. Open `Postman` (or a similar program).
7. We will create a GET request to `http://localhost:18000/login_refresh` with a bearer token authentication header. Paste the token fetch in the previous steps.
8. Send the request. It should succeed.

## Other information

The introduced logic is in the context of a PoC. We will need to review this implementation in a wider context.
